### PR TITLE
CB-11023 New edit-config tag in plugin.xml

### PR DESCRIPTION
**Pull request by ktop**
Documenting the new edit-config tag in plugin.xml. 
@riknoll can you review? I tried to keep the examples short, but explain enough to get the point across. Let me know if I need to fix anything.

### CODE REVIEW

**Comment by macdonst** [comment]
@ktop looks good but my personal preference would be to show a before and after of the Android Manifest.xml you use in your example to really hammer home how it works.

**Comment by riknoll** [comment]
For the examples, I would just give a snippet of the AndroidMainfest and not the whole thing. There's a lot that doesn't change in the before and after. Secondly, the example of force adding plugins is pretty good, but I wonder if this is the right place for it. Perhaps it is better suited for the CLI guide as a sub-heading under the [plugin command](https://github.com/apache/cordova-cli/blob/master/doc/readme.md#cordova-plugin-command). People who get CLI errors might be more likely to look there since they are probably not plugin authors themselves.

**Comment by ktop** [comment]
I shortened the AndroidManifest to snippets. 

I could see the force add content going into both locations, but in the CLI guide it should be more describing the error about conflicting edit-config and how it is caused. Maybe this section can link to the plugin.xml guide about edit-config for how to resolve the error. It's probably worth it for users to learn about edit-config in the plugin.xml guide before making changes.

File: www/docs/en/dev/plugin_ref/spec.md [hunk 1]
```diff
@@ -365,6 +365,111 @@ For windows-specific attributes:
...
+    ...
+        <activity android:configChanges="orientation|keyboardHidden" android:label="NewLabel" android:name="MainActivity">
+            ...
+        </activity>
+    ...
+    <uses-sdk android:maxSdkVersion="23" android:minSdkVersion="16" android:targetSdkVersion="23" />
+</manifest>
+```
+
+Multiple plugins will not be able to to modify the same attributes because it may cause problems with the application. An error will be thrown and plugin install will fail. The conflicting `edit-config` tags must be resolved before the plugin can be added. Make modifications to conflicting tags to resolve the conflict, then remove and re-add those updated plugins.
```
Comment (riknoll) [inline www/docs/en/dev/plugin_ref/spec.md]: "Multiple plugins will not be able to to modify " => "Multiple plugins can not modify"

"then remove and re-add those updated plugins" => "then remove and re-add the updated plugins"

File: www/docs/en/dev/plugin_ref/spec.md [hunk 2]
```diff
@@ -365,6 +365,111 @@ For windows-specific attributes:
...
+
+There is an option for those who are certain that the plugin should be installed despite the conflicts. The `--force` flag can be used with `cordova plugin add`. Force adding the plugin will revert conflicting changes of other plugins so that it can be added without issues. `--force` should be used with caution as reverting changes of other plugins may cause the application to not work as expected.
+
+If the plugins ever get in a weird state, remove all plugins and re-add them.
+
+Example:
```
Comment (riknoll) [inline www/docs/en/dev/plugin_ref/spec.md]: Maybe make this a sub-header?

File: www/docs/en/dev/plugin_ref/spec.md [hunk 3]
```diff
@@ -365,6 +365,111 @@ For windows-specific attributes:
...
+    ...
+        <activity android:configChanges="orientation|keyboardHidden" android:label="NewLabel" android:name="MainActivity">
+            ...
+        </activity>
+    ...
+    <uses-sdk android:minSdkVersion="15" android:targetSdkVersion="23" />
+</manifest>
+```
+
+Note: Reverted changes from `--force` are gone for good. They will not reappear after removing the plugin that was force added. If the reverted changes are needed, the associated plugins should be removed and re-added.
```
Comment (riknoll) [inline www/docs/en/dev/plugin_ref/spec.md]: "the associated plugins" => "all associated plugins"

**Comment by riknoll** [comment]
That make sense to me. I added a few more comments, but otherwise LGTM.

**Comment by ktop** [comment]
@riknoll I've made the changes here and will open a PR in the CLI doc to mention the plugin conflicts and link to the plugin.xml doc for reference.

**Comment by riknoll** [comment]
@macdonst any chance you can take over merging this? Been sick and swamped this week and I don't think I'll have a chance to test out the site before merging it in.

**Pull request closed without merging**